communication/comms.py

```diff
@@ -0,0 +1,185 @@
+"""This module contains a Comms class for providing client-server communication between two pokemon."""
+
+import asyncio
+import random
+import pokemon.pokemon as pkmn
+
+
+class Comms(pkmn.ElectricPokemon):
+    """
+    This Comms class inherits from the ElectricPokemon class.
+
+    Adds client-server communication between pokemon, implemented with asyncio and co-routines.
+
+    Attributes
+    ----------
+    name : str
+        Name of pokemon.
+    nickname : str
+        Nickname of pokemon.
+    moves : list
+        A list of 4 pokemon moves.
+    pokemon_type : str
+        Type of pokemon.
+    ip_address : str
+        The IP address as the client or server (default 127.0.0.1).
+    port_no : int
+        The port number to communicate on (default 8765).
+    """
+
+    def __init__(self, name, nickname, pokemon_type, ip_address="127.0.0.1", port_no=8765):
+        """Initialise Comms class object.
+
+        Initialise with the arguments name, nickname and type, and with
+        optional keyward arguments ip_address and port_no.
+
+        Parameters
+        ----------
+        name : str
+            Name of pokemon.
+        nickname : str
+            Nickname of pokemon.
+        pokemon_type : str
+            Type of pokemon.
+        ip_address : str
+            The IP address as the client or server (default 127.0.0.1).
+        port_no : int
+            The port number to communicate on (default 8765).
+        """
+        self.ip_address = ip_address
+        self.port_no = port_no
+        super(Comms, self).__init__(name, nickname, pokemon_type)
+
+    async def msg_tx(self, message, writer):
+        """Co-routine writes message to socket.
+
+        Parameters
+        ----------
+        message : str
+            The message to write to socket.
+        writer :  StreamWriter object
+            Object instance of StreamWriter class for writing to socket.
+        """
+        print(f"Sent: {message!r}")
+        writer.write(message.encode())
+        await writer.drain()
+
+    async def list_tasks(self, writer):  # noqa: D301
```

Comment:
I don't think it's necessary to avoid D301. Use a `r` at the beginning of your docstring, otherwise the `\n` will disappear when you try to print it. This makes it a "raw" string, rather than one that gets parsed in any way before printing.

Also, this function's name doesn't seem to describe what it does too well. Perhaps something like `generate_random_tasks()`?